### PR TITLE
[opencv/4.x.x] fix CMake imported targets + bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **opencv/4.x.x**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Actually, it doesn't work yet. I guess that build modules are called before targets are defined (but it works when the recipe doesn't use components).

**Comment by SpaceIm** [comment]
Let's wait conan 1.33.0 in CI.

**Comment by jgsogo** [comment]
This is very weird. The CI is identifying `35ae47...` as the latest commit when there is a merge afterward. The problem is that using the previous commit there is a conflict with `master` and it fails.

I would like to have a look at this issue and try to understand why this is happening, but I'm not working this weekend 😅  ... if you want to keep working on this PR I think the best/easiest solution is to push an empty commit and the CI should start to use it.

**Comment by SpaceIm** [comment]
Thanks @jgsogo, I've pushed an empty commit

**Comment by SpaceIm** [comment]
Well, it was a typo. So we can create non namespaced imported targets for recipes with components, even before conan 1.33.0 :)

**Comment by SpaceIm** [comment]
@madebr @prince-chrismc @ericLemanissier @uilianries @SSE4  Could you review please?

**Review by uilianries — APPROVED** [review]
LGTM!

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -283,62 +283,33 @@ def package(self):
         if os.path.isfile(os.path.join(self.package_folder, "setup_vars_opencv4.cmd")):
             os.rename(os.path.join(self.package_folder, "setup_vars_opencv4.cmd"),
                       os.path.join(self.package_folder, "res", "setup_vars_opencv4.cmd"))
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_subfolder, self._module_file),
+            {component["target"]:"opencv::{}".format(component["target"]) for component in self._opencv_components}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
```
Comment (prince-chrismc) [inline recipes/opencv/4.x/conanfile.py:292]: I dislike how much this is being copied around, it really need to be added to Conan

**Review by prince-chrismc — APPROVED** [review]

File: recipes/opencv/4.x/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/opencv/4.x/conanfile.py:292]: Do there exist examples of this pattern for creating imported executables?

**Review by madebr — APPROVED** [review]

File: recipes/opencv/4.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py:292]: I tried something in geographiclib but canceled the modification: https://github.com/conan-io/conan-center-index/pull/4295/commits/ec87edc7f023e156699fd4d17afdd14445aa3469

imported executables will be implemented in conan.

**Pull request merged**